addons/wiki/static/dragNDrop.js

```diff
@@ -0,0 +1,341 @@
+var $ = require('jquery');
+var $osf = require('js/osfHelpers');
+var ctx = window.contextVars;
+var Range = ace.require('ace/range').Range;
+
+var getExtension = function(filename) {
+    return /(?:\.([^.]+))?$/.exec(filename)[1];
+};
+var validImgExtensions = ['jpg', 'jpeg', 'png', 'gif', 'bmp'];
+
+var editor = ace.edit('editor');
+
+
+var autoIncrementFileName = function(name, response) {
+    var num = 1;
+    var newName;
+    var ext = getExtension(name);
+    var baseName = name.replace('.' + ext, '');
+
+    rename:
+    while (true) {
+        for (var i = 0; i < response.data.length; i++) {
+            newName = baseName + '(' + num + ').' + ext;
+            if (response.data[i].attributes.name === newName) {
+                num += 1;
+                newName = baseName + '(' + num + ').' + ext;
+                continue rename;
+            }
+        }
+        break;
+    }
+    return newName;
+};
+
+editor.enable = function() {
+    $('#ace-loading-ball').css('display', 'none');
+    editor.container.style.pointerEvents = 'initial';
+    editor.container.style.opacity = 1;
+    editor.renderer.setStyle('disabled', false);
+};
+
+editor.disable = function() {
+    $('#ace-loading-ball').css('display', 'block');
+    editor.container.style.pointerEvents = 'none';
+    editor.container.style.opacity = 0.1;
+    editor.renderer.setStyle('disabled', true);
+};
+
+var localFileHandler = function(files, cm, init, fixupInputArea, newName) {
+    var multiple = files.length > 1;
+    var urls = [];
+    var name;
+    var ext;
+    var num = cm.addLinkDef(init) + 1;
+    var promises = [];
+    var editor = ace.edit('editor');
+    editor.disable();
+    getOrCreateWikiImagesFolder().fail(function() {
+        notUploaded(multiple);
+        editor.enable();
+    }).done(function(path) {
+        if (!!path) {
+            $.each(files, function(i, file) {
+                ext = getExtension(file.name);
+                name = newName ? newName : file.name;
+                if (validImgExtensions.indexOf(ext.toLowerCase()) <= -1) {
+                    $osf.growl('Error', 'This file type cannot be embedded  (' + file.name + ')', 'danger');
+                    editor.enable();
+                } else {
+                    var waterbutlerURL = ctx.waterbutlerURL + 'v1/resources/' + ctx.node.id + '/providers/osfstorage' + encodeURI(path) + '?name=' + encodeURI(name) + '&type=file';
+                    $osf.trackClick('wiki-markdown-drop-image', ctx.node.id, 'dropped-image');
```

Comment:
@Johnetordoff, I'm not very familiar with our event tracking, but the signature for `trackClick` expects a category, action, and label. So in this case:
```
category = 'wiki-markdown-drop-image'
action = the node guid
label = 'dropped-image'
```
I'm wondering if it makes sense to make the category more general. Maybe just "wiki" in case additional event tracking is added to this page? I'd propose:
```
category = 'wiki-page'
action = 'wiki-edit'
label = 'dropped-image'
```
Thoughts? This is by no means a blocker.

Reply:
There's not really a rhyme or reason to event action/category/label I can see. [Here are the google docs on the matter.](https://developers.google.com/analytics/devguides/collection/analyticsjs/events)

If we use the node guid we can get good approximation of how many people use the feature not just the total number of times the feature is used which I think is more in line with what Product wants.

Reply:
It seems like the order of arguments might be mixed up here. There's more docs about the differences between "category", "action", and "label" here: https://support.google.com/analytics/answer/1033068#Anatomy .

Some examples they give:

```
    Category: "Videos"
    Action: "Play"
    Label: "Baby's First Birthday"
```

```
    Category: "Downloads"
    Action: "PDF"
    Label: "/salesForms/orderForm1.pdf"
```

So I think it'd be more correct for us to do

```
Category: "wiki"
Action: "dropped-image"
Label: <node id>
```